waku/v2/node/wakunode2.nim

```diff
@@ -347,6 +369,12 @@ proc mountRelay*(node: WakuNode, topics: seq[string] = newSeq[string](), rlnRela
 
     node.subscribe(topic, handler)
 
+  if rlnRelayEnabled:
```

Comment:
Should wakuRelay be enabled in rlnRelay is?

Reply:
You mean the other way around? I think rln-relay is a special service for the relay protocol, that is why I have added it inside the relay, let me know whether it makes sense

Reply:
I am going to merge the PR for now, and will apply all the changes related to this comment in the subsequent PRs

Reply:
Typo, what I meant is: If rlnRelay protocol is active, should wakuRelay be active too? Or are they mutually exclusive (if-else)?

Reply:
My understanding of rlnRelay is that it wraps the wakuRelay protocol, so you wouldn't have both rlnRelay and vanilla wakuRelay, but I could be off.

Reply:
> Typo, what I meant is: If rlnRelay protocol is active, should wakuRelay be active too? Or are they mutually exclusive (if-else)?

I see your point, they are not mutually exclusive, indeed rlnRelay **CAN** be active conditioned that wakuRelay is also active i.e., rlnRelay ->wakuRelay but wakuRelay -/-> rlnRelay

Reply:
> My understanding of rlnRelay is that it wraps the wakuRelay protocol, so you wouldn't have both rlnRelay and vanilla wakuRelay, but I could be off.

I considered rln-relay as a separate module/service that can be activated on top of the relay protocol. However, in our current design, given that all the peers are treated as one group, they all have to either have rln-relay enabled or not (this is not an individual decision). But if in the future we consider let's say pubsub topic groups, then enabling rln-relay depends on the topic that a peer wants to contribute to and whether in that topic group spam protection is a concern or not